### PR TITLE
[TASKMGR] Fix text cutoff statusbar2 CPU%, improve statusbar3 Mem

### PR DESCRIPTION
**Pull request by JoachimHenze**
## Purpose

see attached before-and-after-pics in the PR.
also fix an accelerator collision in nl-NL.rc

JIRA issue: none

for the before-pics with the text truncation please have a look into https://github.com/reactos/reactos/pull/5949 where I attached pictures of the state before this PR, where you can see the text-cutoff in very many languages for the CPU-column of statusbar, and also that for very many languages the third field in the statusbar with the Memory is not well visible for taskmgrs default size.

For the after state, where the text cutoff is fixed for **all** languages now, I decided to attach only after pics for those languages that now have the longest strings in the CPU-column. The Memory-column is improved for a lot of languages now, more text is visible there now (but for some languages still not all the text without increasing tasknmgrs size).

bg-BG-after.PNG
bg-BG-after

fr-FR-after.PNG
fr-FR-after

he-IL-after.PNG
he-IL-after

tr-TR-after.PNG
tr-TR-after

### CODE REVIEW

**Review by katahiromz — APPROVED** [review]

File: base/applications/taskmgr/lang/fr-FR.rc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// COPYRIGHT: Benoit Plante (aka cmoibenlepro)
+// COPYRIGHT: Benoit Plante (cmoibenlepro)
```
Comment (binarymaster) [inline base/applications/taskmgr/lang/fr-FR.rc:1]: Unrelated change.

**Review by binarymaster — APPROVED** [review]
One nit, otherwise LGTM.

File: base/applications/taskmgr/taskmgr.h [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 #define STATUS_WINDOW	2001
 #define STATUS_SIZE1	85
-#define STATUS_SIZE2	190
+#define STATUS_SIZE2	157 // he-IL.rc determines 72 requirement
```
Comment (HBelusca) [inline base/applications/taskmgr/taskmgr.h:18]: What do you mean by "72 requirement" ?

File: base/applications/taskmgr/taskmgr.h [hunk 3]
```diff
@@ -36,15 +36,14 @@ typedef struct
 	BOOL	HideWhenMinimized;
 	BOOL	Show16BitTasks;
 
-	/* Update speed settings */
-	/* How many half-seconds in between updates (i.e. 0 - Paused, 1 - High, 2 - Normal, 4 - Low) */
+	// 0 - Paused, 1 - High, 2 - Normal, 4 - Low
```
Comment (HBelusca) [inline base/applications/taskmgr/taskmgr.h:39]:
```suggestion
	/* Update speed settings: 0 - Paused, 1 - High, 2 - Normal, 4 - Low */
```

**Comment by HBelusca** [comment]
Here is the minimal size one can resize Windows 2003 taskmgr window:
image

It looks slightly wider than ours (@JoachimHenze please verify?) If so, I would prefer if we could change our minimal size to one similar to Windows, instead of modifying the strings.

**Review by HBelusca — CHANGES_REQUESTED** [review]
See my large comment.

File: base/applications/taskmgr/taskmgr.h (hunk 2 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/taskmgr.h:18]: I mean the minimum needed width of the size2 field. 157 - 85 == 72.

File: base/applications/taskmgr/taskmgr.h (hunk 3 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/taskmgr.h:39]: Commenting the values makes sense, we do both agree to that.
But I don't see the reason for commenting a variable which is named as "UpdateSpeed" with a comment "update speed",
and it doesn't make any sense to name an attribute that is part of a struct named "Settings" with a comment named "settings". Both is redundant, and not helpful.

So I used the same comment here, that we do use in releases/0.4.7 to releases/0.4.14:
image

**Comment by JoachimHenze** [comment]
> Here is the minimal size one can resize Windows 2003 taskmgr window...

I almost missed that larger comment. Well tbh that comparison isn't fair, because we do use a longer string for the memory part than Windows does (e.g. we do dispay also the usage percentage there), and this is a good thing. I do perceive that as a feature.

In retrun we have to compensate the needed space for that in another area. And the CPU usage field is perfect for that, because it has no dynamic parts. All it needs is a fixed size string + " 100%". If that fits, everything is fine.
So I think in this specific case comparing to 2k3sp2 is not valid.

**Comment by JoachimHenze** [comment]
Here is the smallest I can make my XPSP3ger taskmgr:
image
That is 404x447 pixels

And here is the ros one:
image
That is 404x446 pixels.

So the minimal width is the same! You can see also why Windows needs less space for the memory string there. One number less, and no percentage is shown (which is inferior). Ours is a bit more Win7-ish, and displays more helpful info in the memory field.

File: base/applications/taskmgr/taskmgr.h (hunk 2 above)
Comment (HBelusca) [reply base/applications/taskmgr/taskmgr.h:18]: Then this comment sentence needs to be clarified.

Comment (JoachimHenze) [reply base/applications/taskmgr/taskmgr.h:18]: Do you have a specific suggestion? I could also just strip it.

Comment (HBelusca) [inline base/applications/taskmgr/taskmgr.h:18]: Just simply by spelling out in the comment why "157", based on your explanation given in the PR:
```suggestion
#define STATUS_SIZE2	157 // he-IL.rc determines minimum width:  72 == 157 - 85
```

File: base/applications/taskmgr/lang/nl-NL.rc [hunk 4]
```diff
@@ -406,7 +406,7 @@ BEGIN
     IDS_TAB_IOOTHERBYTES "I/O: overige bytes"
     IDS_TAB_COMMANDLINE "Command Line"
     IDS_MENU_SELECTCOLUMNS "&Kolommen selecteren..."
-    IDS_MENU_16BITTASK "16-&bits taken weergeven" //FIXME: accelerator collision &b with ID_OPTIONS_MINIMIZEONUSE
+    IDS_MENU_16BITTASK "&16-bits taken weergeven"
```
Comment (HBelusca) [inline base/applications/taskmgr/lang/nl-NL.rc:409]: If using "t" as the accelerator, would this collide with something else?

File: base/applications/taskmgr/lang/sv-SE.rc [hunk 5]
```diff
@@ -432,7 +432,7 @@ BEGIN
     IDS_MSG_WARNINGCHANGEPRIORITY "VARNING: Ändring av prioritetsklassen hos den här processen kan\norsaka oönskade effekter och påverka systemets stabilitet. Är du\nsäker på att du vill ändra prioritetsklassen?"
     IDS_MSG_TRAYICONCPUUSAGE "Processoranvändning: %d%%"
     IDS_STATUS_MEMUSAGE "Minnesanv: %s / %s (%d%%)"
-    IDS_STATUS_CPUUSAGE "Processoranv: %3d%%"
```
Comment (HBelusca) [inline base/applications/taskmgr/lang/sv-SE.rc:435]: This is too long?!

File: base/applications/taskmgr/lang/zh-CN.rc [hunk 6]
```diff
@@ -433,7 +433,7 @@ BEGIN
     IDS_MSG_WARNINGCHANGEPRIORITY "警告：改变这一进程的优先级可能会\n导致意外的结果，包括系统不稳定。 您确定\n要更改优先级类？"
     IDS_MSG_TRAYICONCPUUSAGE "CPU 使用情况：%d%%"
     IDS_STATUS_MEMUSAGE "内存使用情况：%s / %s（%d%%）"
-    IDS_STATUS_CPUUSAGE "CPU 使用情况：%3d%%"
+    IDS_STATUS_CPUUSAGE "CPU: %3d%%"
```
Comment (HBelusca) [inline base/applications/taskmgr/lang/zh-CN.rc:436]: Restoring the specific chinese colons:
```suggestion
    IDS_STATUS_CPUUSAGE "CPU： %3d%%"
```

File: base/applications/taskmgr/lang/zh-HK.rc [hunk 7]
```diff
@@ -438,7 +438,7 @@ BEGIN
     IDS_MSG_WARNINGCHANGEPRIORITY "警告：更改此處理程序的優先順序可能導致系統不穩定。\n你確定要更改優先順序嗎？"
     IDS_MSG_TRAYICONCPUUSAGE "CPU 使用情況： %d%%"
     IDS_STATUS_MEMUSAGE "記憶體使用情況： %s ／ %s (%d%%)"
-    IDS_STATUS_CPUUSAGE "CPU 使用情況： %3d%%"
+    IDS_STATUS_CPUUSAGE "CPU: %3d%%"
```
Comment (HBelusca) [inline base/applications/taskmgr/lang/zh-HK.rc:441]:
```suggestion
    IDS_STATUS_CPUUSAGE "CPU： %3d%%"
```

File: base/applications/taskmgr/lang/zh-TW.rc [hunk 8]
```diff
@@ -438,7 +438,7 @@ BEGIN
     IDS_MSG_WARNINGCHANGEPRIORITY "警告：更改此處理程序的優先順序可能導致系統不穩定。\n您確定要更改優先順序嗎？"
     IDS_MSG_TRAYICONCPUUSAGE "CPU 使用情況： %d%%"
     IDS_STATUS_MEMUSAGE "記憶體使用情況： %s ／ %s (%d%%)"
-    IDS_STATUS_CPUUSAGE "CPU 使用情況： %3d%%"
+    IDS_STATUS_CPUUSAGE "CPU: %3d%%"
```
Comment (HBelusca) [inline base/applications/taskmgr/lang/zh-TW.rc:441]:
```suggestion
    IDS_STATUS_CPUUSAGE "CPU： %3d%%"
```

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: base/applications/taskmgr/taskmgr.h (hunk 2 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/taskmgr.h]: Resolved

File: base/applications/taskmgr/taskmgr.h [hunk 9]
```diff
@@ -15,7 +15,7 @@
 
 #define STATUS_WINDOW	2001
 #define STATUS_SIZE1	85
-#define STATUS_SIZE2	190
+#define STATUS_SIZE2	157 // he-IL.rc determines minimum width:  72 == 157 - 85
```
Comment (JoachimHenze) [inline base/applications/taskmgr/taskmgr.h:18]:
```suggestion
#define STATUS_SIZE2	157 // he-IL.rc determines minimum width: 72 == 157 - 85
```
now reactosfanboy-nimize the hermification

File: base/applications/taskmgr/lang/sv-SE.rc (hunk 5 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/lang/sv-SE.rc:435]: yes, it is.

File: base/applications/taskmgr/lang/nl-NL.rc (hunk 4 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/lang/nl-NL.rc:409]: I cannot say, but I am 100% sure, that the &1 won't collide with anything else in that menu (some other languages do use &1 as well)

File: base/applications/taskmgr/lang/zh-HK.rc (hunk 7 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/lang/zh-HK.rc:441]: zh-ZK: Hermes suggested the chinese colon here (although it is not structurally used everywhere else in this file).

File: base/applications/taskmgr/lang/nl-NL.rc (hunk 4 above)
Comment (HBelusca) [reply base/applications/taskmgr/lang/nl-NL.rc:409]: My concern is _also_ about having the keyboard accelerator on a "thin" letter.
See https://learn.microsoft.com/en-us/previous-versions/windows/desktop/dnacc/guidelines-for-keyboard-user-interface-design#selecting-access-keys
> Make access keys discoverable.
>
>- Access keys must be easy for the user to find. Choose letters for access keys as close to the beginning of the label as possible for example, Settings. The following are good letter choices:
    * Letters at the beginning of the first or second word of the label.
    * A distinctive consonant or a vowel in the label.
    * _**Letters with wide widths, such as w, m, and capital letters.**_
>
>- Avoid using letters that are difficult to see when they are assigned to access keys. These letters include:
    * Letters with descenders, such as p, g, q, or y.
    * Letters next to a letter with a descender.
    * _**Letters that are only one pixel wide, such as i or l.**_

**Review by HBelusca — APPROVED** [review]
OK, but consider my spiel about the keyboard accelerator on "1" not being a good thing, before merging.

File: base/applications/taskmgr/lang/nl-NL.rc (hunk 4 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/lang/nl-NL.rc:409]: Wise words from MS. Wise words from you.
Nevertheless I do think the &1 is not as bad as &i. It is considerably less "thin". I do think the result is visually appealing, see the pic:
image

&1 is also used in some other langs, so I don't see using it here as well as a big no-no.
Best ofc would be knowing what MS actually uses here for nl-NL, but I don't have a netherlandish Windows version to find that out for sure.
So I decided to stick with &1 for now **to solve the existing collision at least**, that we had here.
To address your concern I decided to add a //FIXME: Please double-check this accelerator
at that line so that a native netherland-er can cover the details later.

Comment (JoachimHenze) [inline base/applications/taskmgr/lang/nl-NL.rc:409]:
```suggestion
    IDS_MENU_16BITTASK "&16-bits taken weergeven" //FIXME: Please double-check this accelerator
```

File: base/applications/taskmgr/lang/fr-FR.rc (hunk 1 above)
Comment (JoachimHenze) [reply base/applications/taskmgr/lang/fr-FR.rc:1]: Unrelated and shorter: yes.
Wrong: no.

Comment (binarymaster) [reply base/applications/taskmgr/lang/fr-FR.rc:1]: I'd prefer such changes in a separate PR and throughout entire source tree, fixing all occurences, i.e. not just because being "while at it"./

Comment (JoachimHenze) [reply base/applications/taskmgr/lang/fr-FR.rc:1]: There is just one additional hit in the whole source-tree, if you want you can create a PR to remove it there as well:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=aka+cmoibenlepro%29
I don't want it to be part of this PR, and I don't want to exclude my change from this PR neither.

Comment (binarymaster) [reply base/applications/taskmgr/lang/fr-FR.rc:1]: What I meant is the `"%Full_Name% (aka %nickname%)"` pattern.

Comment (JoachimHenze) [reply base/applications/taskmgr/lang/fr-FR.rc:1]: Like I said: Feel free to open a PR for it. But please stop nitpicking in mine about that comment-change. I will not undo that change.

File: base/applications/taskmgr/lang/nl-NL.rc (hunk 4 above)
Comment (HBelusca) [reply base/applications/taskmgr/lang/nl-NL.rc]: OK no problem then, if we already did that in other languages. But this also depends on the user GUI font: if it doesn't print the 1 with an extra horizontal bar at the bottom (the serif) and thus, its width be further reduced then the keyboard accelerator symbol may not be seen greatly.
Also one may say that from some viewing distance, the keyboard accelerator bar may be confounded with the serif.
